### ISSUE TITLE
Unnecessary classifier retraining when using python API

### ISSUE THREAD

**Issue by clbarnes** [ilastik/ilastik#2134]
See here for my initial report: https://forum.image.sc/t/ilastik-from-python-processing-more-than-1-image-starts-retraining/28146

## Describe the bug

Loading a fully-trained classifier (2-stage autocontext in my case) and using it to process a preloaded array works fine. However, if a second array is processed (either by making a second call to `batchProcessingApplet.run_export`, or by adding a second `DatasetInfo` to the first), the classifier is retrained. This happens even when the project is set to `readonly`.

## Expected behavior

Classifiers should not need to retrain if nothing changes about the training data or classifier structure: large amounts of data should be process-able from python.

## To Reproduce

1. Train a 2-stage autocontext ILP over a dataset in `xytc` and save it

```python
from collections import OrderedDict

import vigra
import numpy as np
import ilastik_main
from ilastik.applets.dataSelection import DatasetInfo

args = ilastik_main.parse_args([])
args.headless = True
args.readonly = True
args.project = "path/to/autocontext2.ilp"

shell = ilastik_main.main(args)

arrays = [
    vigra.taggedView(
        np.random.randint(0, 256, (100, 100, 10, 1), dtype=np.uint8),
        "xytc"
    ) for _ in range(2)
]


def two_arrays_at_once():
    role_data_dict = OrderedDict([
        ("Raw Data", [DatasetInfo(preloaded_array=a) for a in arrays])
    ])

    return shell.workflow.batchProcessingApplet.run_export(
        role_data_dict, export_to_array=True
    )


def one_then_another():
    role_data_dict1 = OrderedDict([
        ("Raw Data", [DatasetInfo(preloaded_array=arrays[0])])
    ])

    preds1 = shell.workflow.batchProcessingApplet.run_export(
        role_data_dict1, export_to_array=True
    )

    role_data_dict2 = OrderedDict([
        ("Raw Data", [DatasetInfo(preloaded_array=arrays[1])])
    ])

    preds2 = shell.workflow.batchProcessingApplet.run_export(
        role_data_dict2, export_to_array=True
    )

    return preds1 + preds2


if __name__ == '__main__':
    two_arrays_at_once()
    # one_then_another()
```

## Desktop (please complete the following information):

 - ilastik version: e.g. 1.3.3b1
- OS: Ubuntu 16.04

**Comment by clbarnes** [comment]
@k-dominik took a look at this in response to the forum post.

**Comment by k-dominik** [comment]
fixed in #2375
